### PR TITLE
chore(trace): move span module

### PR DESCRIPTION
**Pull request by majorgreys**
Stacked after #8234 and #8251.

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/internal/ci_visibility/encoder.py [hunk 1]
```diff
@@ -199,7 +199,7 @@ def _convert_span(self, span, dd_origin):
         converted_span = {
             "test_session_id": int(span.get_tag(SESSION_ID) or "1"),
             "test_suite_id": int(span.get_tag(SUITE_ID) or "1"),
-            "files": json.loads(span.get_tag(COVERAGE_TAG_NAME))["files"],
+            "files": json.loads(span.get_tag(COVERAGE_TAG_NAME) or "{}")["files"],
```
Comment (romainkomorndatadog) [inline ddtrace/internal/ci_visibility/encoder.py:202]: Doesn't this automatically cause a`KeyError` if we hit the `or`?

Comment (romainkomorndatadog) [reply ddtrace/internal/ci_visibility/encoder.py:202]: Not that the original version is much better, though. I should probably just rewrite this.

Comment (majorgreys) [reply ddtrace/internal/ci_visibility/encoder.py:202]: Glad you caught this, I had to add it because of a typing error:

```
ddtrace/internal/ci_visibility/encoder.py:202: error: Argument 1 to "loads" has incompatible type "Optional[str]"; expected "Union[str, bytes]"  [arg-type]
```

I see your point about the `KeyError` and changed it to cast the return value of `get_tag` to a string.

**Review by P403n1x87 — APPROVED** [review]

**Review by Yun-Kim — APPROVED** [review]
looks good on mlobs side

**Review by gnufede — APPROVED** [review]

**Review by tabgok — APPROVED** [review]

**Review by ZStriker19 — APPROVED** [review]

File: ddtrace/internal/ci_visibility/encoder.py [hunk 2]
```diff
@@ -199,7 +199,7 @@ def _convert_span(self, span, dd_origin):
         converted_span = {
             "test_session_id": int(span.get_tag(SESSION_ID) or "1"),
             "test_suite_id": int(span.get_tag(SUITE_ID) or "1"),
-            "files": json.loads(span.get_tag(COVERAGE_TAG_NAME))["files"],
+            "files": json.loads(str(span.get_tag(COVERAGE_TAG_NAME)))["files"],
```
Comment (emmettbutler) [inline ddtrace/internal/ci_visibility/encoder.py:202]: How is this related to moving the module?

**Review by emmettbutler — CHANGES_REQUESTED** [review]
The change looks good. I believe you need to add `ddtrace/_trace` to the suitespec to make sure tests continue to be triggered after this change.

File: ddtrace/internal/ci_visibility/encoder.py (hunk 2 above)
Comment (majorgreys) [reply ddtrace/internal/ci_visibility/encoder.py:202]: https://github.com/DataDog/dd-trace-py/pull/8274#discussion_r1482034544

Comment (majorgreys) [reply ddtrace/internal/ci_visibility/encoder.py:202]: Better to leave a comment in code?

Comment (emmettbutler) [reply ddtrace/internal/ci_visibility/encoder.py:202]: Nah it's a reasonable change - I just had a hard time understanding how it's related.

**Review by emmettbutler — APPROVED** [review]
